(SOS My Discord Crashed!) Update BirdAI: now chasing the closest target

WendyFu004 opened the pull request:
--add "AI" tag for AI charactors so the bird could find the closest AI from this tag as the target
--change tag "AI" of race track to "RaceTrack"
Now the birds will chasing the closest Ai within the target radius
My discord looping update failed !!! Sorry I can't use discord for now 
I am so SORRY X(

WendyFu004 committed:
Birds now chasing the closest target

--add "AI" tag for AI charactors so the bird could find the closest AI from this tag as the target
--change tag "AI" of race track to "RaceTrack"
Now the birds will chasing the closest Ai within the target radius

linchen2508 reviewed (changes requested):
Since the birds try to hit the rockets as hard as possible, i think we should use pursue or seek instead of arrive, also the speed could be made a little bit faster

WendyFu004 committed:
Rework bird AI

Assign each bird a list of racers inside the restricted area and the area center.
If the distance between bird and racer is less then the BireRetargetRadius the target will change and also add an collision detection to the bird.

linchen2508 committed:
Fix the 3 birds sticking together problem

Implemented the fix for the issue where 3 birds stick together after all the racers are out of range

WendyFu004 committed:
Rework BirdAI by using external BirdAIManager

Add bird AI manager to the restricted area center to make the decision for the AI birds.

WendyFu004 committed:
Merge branch 'BirdAI' of https://github.com/afourneaux/rocket-racing-476 into BirdAI

WendyFu004 committed:
Solved compiler error

-

linchen2508 reviewed (changes requested):
The bird takes a very long time to go back to the original position after the racers got out of range
